azurerm/data_source_netapp_account.go

```diff
@@ -0,0 +1,108 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	aznetapp "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/netapp"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetAppAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetAppAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: aznetapp.ValidateNetAppAccountName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"active_directory": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dns_servers": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"smb_server_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"username": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"password": {
+							Type:      schema.TypeString,
+							Sensitive: true,
+							Computed:  true,
+						},
+						"organizational_unit": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceArmNetAppAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.AccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: NetApp Account %q (Resource Group %q) was not found", name, resourceGroup)
+		}
+		return fmt.Errorf("Error reading NetApp Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.AccountProperties; props != nil {
+		if err := d.Set("active_directory", flattenArmNetAppActiveDirectories(props.ActiveDirectories)); err != nil {
+			return fmt.Errorf("Error setting `active_directory`: %+v", err)
+		}
+	}
+
+	currentTags := make(map[string]*string)
+	if v := resp.Tags; v != nil {
+		tagMap := v.(map[string]interface{})
+
+		for k, v := range tagMap {
+			currentTags[k] = utils.String(v.(string))
+		}
+	}
+
+	return tags.FlattenAndSet(d, currentTags)
```

Comment:
I think this can be
```suggestion
	return tags.FlattenAndSet(d, resp.Tags)
```

Reply:
It seems that we cannot do this since the type of resp.Tags is "Tags interface{} json:"tags,omitempty"" and function tags.FlattenAndSet only can accept type "tagMap map[string]*string". If I write code as "return tags.FlattenAndSet(d, resp.Tags)" and it would throw below error message. So we need this.
![image](https://user-images.githubusercontent.com/19754191/66373739-ecb67180-e9db-11e9-936b-b6b5f922e4ba.png)

Reply:
thanks for the clarification! makes sense